EIPS/eip-2537.md

```diff
@@ -0,0 +1,310 @@
+---
+eip: 2537
+title: Precompile for BLS12-381 curve operations
+author: Alex Vlasov (@shamatar)
+discussions-to: https://ethereum-magicians.org/t/eip2537-bls12-precompile-discussion-thread/4187
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-02-21
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+
+This precompile adds operation on BLS12-381 curve as a precompile in a set necessary to *efficiently* perform operations such as BLS signature verification and perform SNARKs verifications.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+If `block.number >= X` we introduce *nine* separate precompiles to perform the following operations (addresses to be determined):
```

Comment:
Can we get notional contract addresses written into the spec?  I believe the go implementation has proposed `0xA`->`0x12` for these.

Reply:
@shamatar friendly ping :-)

Reply:
Addresses are up in the last commit